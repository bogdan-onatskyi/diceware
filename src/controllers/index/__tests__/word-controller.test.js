import React from 'react';

import WordController from '../word-controller';
import Word from '../../../models/word-object';

describe("WordController component", () => {
    it("should match saved snapshot (wordObject.editorOpened === false)", () => {
        const wordObject = new Word(1);
        wordObject.index = 0;

        const wrapper = shallow(
            <WordController wordObject={wordObject}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should match saved snapshot (wordObject.editorOpened === true)", () => {
        const wordObject = new Word(1);
        wordObject.index = 0;
        wordObject.editorOpened = true;

        const wrapper = shallow(
            <WordController wordObject={wordObject}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it("should have 1 prop", () => {
        const propsStr = 'wordObject';
        const props = [];
        for (let prop in WordController.propTypes) props.push(prop);
        expect(props.join(' ') === propsStr).toBe(true);
    });
});