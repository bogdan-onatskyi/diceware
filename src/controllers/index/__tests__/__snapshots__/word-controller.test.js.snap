// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordController component should match saved snapshot (wordObject.editorOpened === false) 1`] = `
<div
  className="w-container"
>
  <WordView
    code="11111"
    filter=""
    handleClick={[Function]}
    handleWheel={[Function]}
    next1word="ABDOMEN"
    next2word="ABDOMINAL"
    prev1word="ZOOM"
    prev2word="ZOOLOGY"
    toggleEditor={[Function]}
    word="ABACUS"
    wordViewId={2}
  />
</div>
`;

exports[`WordController component should match saved snapshot (wordObject.editorOpened === true) 1`] = `
<div
  className="w-container"
>
  <WordView
    code="11111"
    filter=""
    handleClick={[Function]}
    handleWheel={[Function]}
    next1word="ABDOMEN"
    next2word="ABDOMINAL"
    prev1word="ZOOM"
    prev2word="ZOOLOGY"
    toggleEditor={[Function]}
    word="ABACUS"
    wordViewId={2}
  />
  <EditorDialog
    filter=""
    handleBackspace={[Function]}
    handleCountWords={[Function]}
    handleFilter={[Function]}
    toggleEditor={[Function]}
    wordView={
      <WordView
        code="11111"
        filter=""
        handleClick={[Function]}
        handleDoubleClick={undefined}
        handleWheel={[Function]}
        next1word="ABDOMEN"
        next2word="ABDOMINAL"
        prev1word="ZOOM"
        prev2word="ZOOLOGY"
        toggleEditor={[Function]}
        word="ABACUS"
        wordViewId={2}
      />
    }
  />
</div>
`;
